backend-config/namespace_config.go

```diff
@@ -105,6 +119,10 @@ func (nc *namespaceConfig) getFromAPI(ctx context.Context) (map[string]ConfigT,
 		respBody = configEnvHandler.ReplaceConfigWithEnvVariables(respBody)
 	}
 
+	if !useUpdateAfter {
+		nc.workspacesConfig = make(map[string]ConfigT)
```

Comment:
This allocation would be unnecessary in case we return earlier due to an unmarshal error on 127.

We should move it right before the last `for` loop.

One question though, why do we initialize this map only if `!useUpdateAfter`? Shouldn't it simply be `if nc.workspaceConfig == nil {`?

Reply:
(1) Yes, changed the location.
(2) No, we can't do that because if useUpdateAfter is false, we should be rebuilding the entire map (to keep the semantics of not using useUpdateAfter exactly the same as before

Reply:
I see. Clever although I find it confusing because you're relying on `IsZero()` above. It's not obvious at a first glance.

Personally I'd find it more immediate if the map is initialized in the constructor (i.e. `SetUp`) as it's quite expected/common thing to do. Then you could just change the `if !useUpdateAfter` into:

```go
if !nc.useIncrementalConfigUpdates {
	nc.workspacesConfig = make(map[string]ConfigT)
}
```

Not mandatory of course, just a suggestion to make the code easier to understand.

Reply:
Done!